textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/reader/LuceneDataReader.java

```diff
@@ -44,14 +40,11 @@
     private IndexSearcher indexSearcher;
     private ScoreDoc[] scoreDocs;
     private IndexReader indexReader;
-    private String query;
-    private String defaultField;
+    private Query query;
     
-    public LuceneDataReader(IDataStore dataStore, 
-            String query, String defaultField) {
+    public LuceneDataReader(IDataStore dataStore, Query query) {

```

Comment:
Lucene can't make sure the analyzer in reader is consistent with the analyzer in writer. The interface interacting with Lucene should take care of that. If the analyzers are not consistent we might end up getting unexpected results.
For example, if we have used `StandardAnalyzer` while indexing and `WhiteSpaceAnalyzer` while querying the results may not be consistent with the expected results.